src/main.rs

```diff
@@ -90,7 +109,62 @@ fn app(pool: PgPool) -> Router {
         )
         .fallback(not_found_handler);
 
-    router.with_state(pool)
+    router.with_state(state)
+}
+
+// TODO: Clean this shitshow of a function up
+// There is unwrap and clone everywhere
+// We can't remove everything but god damn.
+// TODO: Should we use the request body as a cache key as well?
```

Comment:
Why would u use the request body as a cache key as well?

Reply:
Because it is weird if two different requests with different bodies, but same url, method and idempotency keys returns the same thing

Reply:
Why would they have the same idempotency keys? Also, as  idempotency keys would GUIDs not be better?

Reply:
Yes but remember it gets hashed we don't actually store all that information. Generating idemptency keys is the responsibility of the client so they should generate a strong key (yes that could be a guid). It is just to mitigate errors where an idempotency key is reused across two unrelated requests. For example as a result of a programming error.

Reply:
I don't think we need to consider programming errors.. they can just git gud..